include/vcpkg/metrics.h

```diff
@@ -22,18 +84,23 @@ namespace vcpkg
 
         void track_metric(const std::string& name, double value);
         void track_buildtime(const std::string& name, double value);
-        void track_property(const std::string& name, const std::string& value);
-        void track_property(const std::string& name, bool value);
+
+        void track_property(DefineMetric metric);
+        void track_property(StringMetric metric, const std::string& value);
+        void track_property(BoolMetric metric, bool value);
+
         void track_feature(const std::string& feature, bool value);
-        void track_option(const std::string& option, bool value);
 
         bool metrics_enabled();
 
         void upload(const std::string& payload);
         void flush(Filesystem& fs);
-    };
 
-    Optional<StringView> find_first_nonzero_mac(StringView sv);
+        // exposed for testing
```

Comment:
There are exposed for testing, but I see no tests.

Reply:
tests have been added